packages/misc/src/lib/misc.ts

```diff
@@ -144,11 +144,46 @@ export const throwErrorV0 = ({
 };
 
 // Map for old error codes to new ones
-const oldErrorToNewErrorMap: { [key: string]: string } = {
+const oldErrorToNewErrorMap: Record<string, string> = {
   not_authorized: 'NodeNotAuthorized',
   storage_error: 'NodeStorageError',
 };
 
+class GenericError extends Error {
```

Comment:
1. So there's a bit of a weird situation here, because we've been `throw`ing a lot of plain javascript objects which serialise to JSON -- but the native Error object _doesn't_. Note that if you `console.log(JSON.stringify(new Error('wat')));` you'll see `{}` is the output. This is because standard Errors don't include a toJSON -- historically, this was because the `stack` property is 'special'.

So in order to keep existing serialisation behaviour somewhat close (e.g. people will check if JSON.stringify(err) `contains` a string, but `{}` never will contain any of the strings they are checking, it would be good to go ahead and make these errors extend from `openagenda/verror`, which defines a solid `toJSON()` implementation.

For context for the reader:
- VError is Joyent's canonical error package for NodeJS.
- `openagenda/verror` is a browser-friendly version of `verror` that is much smaller than verror and has no dependencies on nodeJS internals.

2. Note that `@openagenda/verror` supports injecting any number of properties without a customized constructor signature, by providing `meta` in the constructor; this way constructing our errors will be consistent everywhere.

3. `GenericError` is not that generic; it actually is quite opinionated that it will have a requestId, for example.  Let's call it `LitNodeRequestError` instead for clarity?

4. `LitNodeRequestErrorV1` would be something like:
```javascript
class LitNodeRequestErrorV1 extends LitNodeRequestError {
  readonly status?: number
  readonly details?: string[]
}
```

Constructing it would be:
```
throw new LitNodeRequestErrorV1({
  name: 'LitNodeRequestErrorV1'
  meta: {
    requestId,
    errorKind,
    errorCode,
    status,
    details,
  },
  constructorOpt: throwErrorV1
});
```

Note:
- The use of `constructorOpt` will eliminate the `throwErrorV1` function from the stack trace on the error
- `meta` keys are set on the error as properties without needing to make a custom constructor signature. We can use a function for that purpose, and avoid constructor signature mismatches and `super()` order of execution irritations
- To see examples of how to subclass VError with pre-set values injected for `meta`, but supporting the caller to also override those, and ensuring the entire VError constructor API is maintained, check out this example where http errors are being shipped in `@openagenda/verror`: https://github.com/OpenAgenda/verror/blob/master/src/http.js#L8

Reply:
This is solved with https://github.com/LIT-Protocol/js-sdk/pull/576

As the plan is to merge and release v7 now, I'm closing this PR. The other one is way better and this one will only create conflicts